PythonVisualizations/Array.py

```diff
@@ -197,19 +219,25 @@ def createCellValue(self, indexOrCoords, key, color=None):
         return cell_rect, cell_val
 
     def display(self):
+        callEnviron = self.createCallEnvironment()
         self.canvas.delete("all")
 
         for i in range(self.size):  # Draw grid of cells
             self.createArrayCell(i)
+            
+        # draw an index pointing to the last item in the list
+        indexDisplay = self.createIndex(len(self.list)-1, "nItems", level = -1, color = 'black')
+        callEnviron |= set(indexDisplay)        

```

Comment:
It's a good idea to create the `nItems` index in the `display()` method since it used by the data structure and all the operations.  Three important changes are needed:
 * `nItems` always points at the (empty) cell following the last filled cell.  That's because it starts at 0 (since the number of items is 0) which is the first cell of the array.  So instead of `len(self.list)-1`, the initial position should be `len(self.list)`.
 * There's no need to add the `nItems` index to the call environment because it does not need to be deleted; it stays on the canvas with the array cells across all the operations.  So line 230 is not needed.
 * The other operations will need to manipulate the `nItems` index, so it should be stored in a instance attribute of the object.  That will allow the `insert()`, `delete()`, etc. methods to change it.